File: examples/companion-packages/zig-zon-app/src/pets.zig
//! Static pet data shared across the CLI commands.

const std = @import("std");

/// Number of days in shelter at which a pet is considered "long-stay".
pub const long_stay_threshold: u32 = 120;

pub const Pet = struct {
    slug: []const u8,
    name: []const u8,
    species: []const u8,
    breed: []const u8,
    age_years: u32,
    days_in_shelter: u32,
    tags: []const []const u8,
    needs: []const u8,
};

/// Mirrors the rust-cargo-app data so companion skills see the same
/// fictional shelter in every ecosystem port.
pub const all_pets = [_]Pet{
    .{
        .slug = "biscuit",
        .name = "Biscuit",
        .species = "dog",
        .breed = "Beagle mix",
        .age_years = 4,
        .days_in_shelter = 12,
        .tags = &.{ "playful", "social", "good-with-kids" },
        .needs = "Daily walks; loves squeaky toys.",
    },
    .{
        .slug = "pepper",
        .name = "Pepper",
        .species = "cat",
        .breed = "Domestic Shorthair",
        .age_years = 8,
        .days_in_shelter = 247,
        .tags = &.{ "mellow", "lap-cat", "solo" },
        .needs = "Quiet home preferred; no other cats.",
    },
    .{
        .slug = "marigold",
        .name = "Marigold",
        .species = "dog",
        .breed = "Senior Labrador",
        .age_years = 11,
        .days_in_shelter = 89,
        .tags = &.{ "calm", "gentle", "low-energy" },
        .needs = "Joint supplements; short walks only.",
    },
    .{
        .slug = "tofu",
        .name = "Tofu",
        .species = "rabbit",
        .breed = "Holland Lop",
        .age_years = 2,
        .days_in_shelter = 31,
        .tags = &.{ "curious", "social" },
        .needs = "Roomy enclosure and unlimited hay.",
    },
    .{
        .slug = "otis",
        .name = "Otis",
        .species = "dog",
        .breed = "Pittie mix",
        .age_years = 5,
        .days_in_shelter = 156,
        .tags = &.{ "gentle", "good-with-kids", "no-cats" },
        .needs = "Cat-free home; loves toddlers.",
    },
    .{
        .slug = "juniper",
        .name = "Juniper",
        .species = "cat",
        .breed = "Tortoiseshell",
        .age_years = 3,
        .days_in_shelter = 22,
        .tags = &.{ "vocal", "spunky", "solo" },
        .needs = "Only cat in the household, please.",
    },
    .{
        .slug = "maple",
        .name = "Maple",
        .species = "dog",
        .breed = "Mini Australian Shepherd",
        .age_years = 1,
        .days_in_shelter = 6,
        .tags = &.{ "high-energy", "smart", "needs-training" },
        .needs = "Training class strongly recommended.",
    },
    .{
        .slug = "clover",
        .name = "Clover & Sage",
        .species = "guinea-pig",
        .breed = "Bonded pair",
        .age_years = 1,
        .days_in_shelter = 18,
        .tags = &.{ "social", "bonded-pair" },
        .needs = "Must adopt together — bonded for life.",
    },
};

pub fn isLongStay(p: *const Pet) bool {
    return p.days_in_shelter >= long_stay_threshold;
}

pub fn findBySlug(slug: []const u8) ?*const Pet {
    for (&all_pets) |*p| {
        if (std.ascii.eqlIgnoreCase(p.slug, slug)) return p;
    }
    return null;
}

pub fn matchesSpecies(p: *const Pet, species: []const u8) bool {
    if (species.len == 0) return true;
    return std.ascii.eqlIgnoreCase(p.species, species);
}

File: examples/companion-packages/golang-app/internal/pawmatch/pets.go
package pawmatch

import "strings"

// LongStayThreshold is the number of days in shelter at which a pet is
// considered "long-stay" and surfaced more prominently to potential adopters.
const LongStayThreshold = 120

// Pet describes an adoptable animal in the example shelter.
type Pet struct {
	Slug          string
	Name          string
	Species       string
	Breed         string
	AgeYears      int
	DaysInShelter int
	Tags          []string
	Needs         string
}

// AllPets is the curated static roster used by the example CLI. Mirrors the
// npm-javascript-app data exactly so the companion skills see the same
// fictional shelter in every ecosystem port.
var AllPets = []Pet{
	{
		Slug:          "biscuit",
		Name:          "Biscuit",
		Species:       "dog",
		Breed:         "Beagle mix",
		AgeYears:      4,
		DaysInShelter: 12,
		Tags:          []string{"playful", "social", "good-with-kids"},
		Needs:         "Daily walks; loves squeaky toys.",
	},
	{
		Slug:          "pepper",
		Name:          "Pepper",
		Species:       "cat",
		Breed:         "Domestic Shorthair",
		AgeYears:      8,
		DaysInShelter: 247,
		Tags:          []string{"mellow", "lap-cat", "solo"},
		Needs:         "Quiet home preferred; no other cats.",
	},
	{
		Slug:          "marigold",
		Name:          "Marigold",
		Species:       "dog",
		Breed:         "Senior Labrador",
		AgeYears:      11,
		DaysInShelter: 89,
		Tags:          []string{"calm", "gentle", "low-energy"},
		Needs:         "Joint supplements; short walks only.",
	},
	{
		Slug:          "tofu",
		Name:          "Tofu",
		Species:       "rabbit",
		Breed:         "Holland Lop",
		AgeYears:      2,
		DaysInShelter: 31,
		Tags:          []string{"curious", "social"},
		Needs:         "Roomy enclosure and unlimited hay.",
	},
	{
		Slug:          "otis",
		Name:          "Otis",
		Species:       "dog",
		Breed:         "Pittie mix",
		AgeYears:      5,
		DaysInShelter: 156,
		Tags:          []string{"gentle", "good-with-kids", "no-cats"},
		Needs:         "Cat-free home; loves toddlers.",
	},
	{
		Slug:          "juniper",
		Name:          "Juniper",
		Species:       "cat",
		Breed:         "Tortoiseshell",
		AgeYears:      3,
		DaysInShelter: 22,
		Tags:          []string{"vocal", "spunky", "solo"},
		Needs:         "Only cat in the household, please.",
	},
	{
		Slug:          "maple",
		Name:          "Maple",
		Species:       "dog",
		Breed:         "Mini Australian Shepherd",
		AgeYears:      1,
		DaysInShelter: 6,
		Tags:          []string{"high-energy", "smart", "needs-training"},
		Needs:         "Training class strongly recommended.",
	},
	{
		Slug:          "clover",
		Name:          "Clover & Sage",
		Species:       "guinea-pig",
		Breed:         "Bonded pair",
		AgeYears:      1,
		DaysInShelter: 18,
		Tags:          []string{"social", "bonded-pair"},
		Needs:         "Must adopt together — bonded for life.",
	},
}

// IsLongStay reports whether a pet has been in the shelter long enough to
// qualify as a long-stay friend.
func IsLongStay(p Pet) bool {
	return p.DaysInShelter >= LongStayThreshold
}

// FindPetBySlug returns the pet with the matching slug (case-insensitive),
// or false if none.
func FindPetBySlug(slug string) (Pet, bool) {
	target := strings.ToLower(slug)
	for _, p := range AllPets {
		if strings.ToLower(p.Slug) == target {
			return p, true
		}
	}
	return Pet{}, false
}

// FilterPetsBySpecies returns the pets matching the given species
// (case-insensitive). An empty species returns all pets.
func FilterPetsBySpecies(species string) []Pet {
	if species == "" {
		out := make([]Pet, len(AllPets))
		copy(out, AllPets)
		return out
	}
	target := strings.ToLower(species)
	out := make([]Pet, 0, len(AllPets))
	for _, p := range AllPets {
		if strings.ToLower(p.Species) == target {
			out = append(out, p)
		}
	}
	return out
}
